### PR TITLE
Signals handling SIGHUP triggers configuration reload, SIGINT gracefu…

### PR DESCRIPTION
**Pull request by mjarco**
…lly shutsdown server

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -52,41 +49,45 @@ func main() {
...
-		log.Fatalf("Improperly configured %s", err)
+		log.Fatalf("Configuration corrupted: %s", err)
 	}
 
-	valid, errs := config.ValidateConf(conf.YamlConfig, true)
-	if !valid {
-		log.Printf("YAML validation - errors: %q", errs)
-		os.Exit(1)
+	if *testConfig {
+		os.Exit(0)
```
Comment (wookie41) [inline main.go]: I don't understand. Why we exit with success when config is defined?

Comment (mjarco) [reply main.go]: It's configuration dry-run ./akubra -c config.yaml -t. Like nginx -t validates changed configs

File: main.go [hunk 2]
```diff
@@ -105,72 +106,125 @@ func mkServiceLogs(logConf logconfig.LoggingConfig) (syncLog, clusterSyncLog, ac
...
-			Handler:      handler,
-			ReadTimeout:  s.config.Service.Server.ReadTimeout.Duration,
-			WriteTimeout: s.config.Service.Server.WriteTimeout.Duration,
-		},
-		Timeout: s.config.Service.Server.ShutdownTimeout.Duration,
-	}
-
-	srv.SetKeepAlivesEnabled(true)
-	listener, err := net.Listen("tcp", s.config.Service.Server.Listen)
+	err = metrics.Init(conf.Metrics)
```
Comment (bieli) [inline main.go]: My proposition it's moving `err = metrics.Init(conf.Metrics)` line outside HUP signal block - I suspect many goroutines on stack after recive many HUP signals.

File: main.go [hunk 3]
```diff
@@ -180,19 +234,13 @@ func (s *service) startTechnicalEndpoint() {
...
-				MaxHeaderBytes: 512,
-				WriteTimeout:   TechnicalEndpointGeneralTimeout,
-				ReadTimeout:    TechnicalEndpointGeneralTimeout,
-			},
-			Timeout:      TechnicalEndpointGeneralTimeout,
-			TCPKeepAlive: 1 * time.Minute,
-			Logger:       graceful.DefaultLogger(),
+		srv := &http.Server{
+			Addr:           port,
+			Handler:        serveMuxHandler,
```
Comment (bieli) [inline main.go]: `port` variable is related to shared config. It will be nice to refresh `http.Server` in  `startTechnicalEndpoint`

**Pull request closed without merging**